config/application.rb

```diff
@@ -38,7 +38,11 @@ class Application < Rails::Application
 
     config.middleware.insert_before 0, Rack::Cors do
       allow do
-        origins '*'
+        origins(lambda do |source, _env|
+          ServiceProvider.pluck(:redirect_uris).flatten.map do |uri|
```

Comment:
Is this going to scan the entire column in the database for every single request? or only for CORS requests?

Reply:
It will call this code on every request that contains `HTTP_ORIGIN` header. The column is very small, though. Will this be an issue?

Also, I think there's an issue with this code. I was able to get a 200 response and see the JSON response when setting the origin to a value that isn't whitelisted, such as:
```
curl -H "Origin: foo" http://localhost:3000/.well-known/openid-configuration
```
Shouldn't this return a 404 or some other non-200 code?

Reply:
No, CORS is not about changing status codes or the body of the request, just the presence of these specific headers. CORS is respected by browsers and adding these headers allows AJAX requests from other domains

Reply:
The CORS headers are "silent" if the conditions aren't met:

```
# allowed origin -- has headers
> curl -v -H "Origin: http://localhost:9292" http://localhost:3000/.well-known/openid-configuration 2>&1 | grep Access
< Access-Control-Allow-Origin: http://localhost:9292
< Access-Control-Allow-Methods: GET
< Access-Control-Max-Age: 1728000
# unknown origin -- no headers
> curl -v -H "Origin: foo" http://localhost:3000/.well-known/openid-configuration 2>&1 | grep Access
> 
```

Reply:
Makes sense, thanks. Typical requests will not contain the `HTTP_ORIGIN` header, so we should be good with the DB calls, right?

Reply:
Confirmed! I just added some logging & tested locally